airflow/jobs.py

```diff
@@ -2619,14 +2591,14 @@ def on_kill(self):
 
     @provide_session
     def heartbeat_callback(self, session=None):
-        """Self destruct task if state has been moved away from running externally"""
+        """Reset state to running if state set to null, scheduled, or queued externally.  Self-destruct if state is set to any other state externally."""
```

Comment:
This is the beginning of the "old worker" side of the scheduler bounce recovery scenario.  This is in `LocalTaskJob.heartbeat_callback()` which is responsible for responding to external mutations to the task's state.